code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -99,6 +100,11 @@ public Account createAccount(String name, String kind, String externalId, String
     }
 
     @Override
+    public void deleteAccount(Account account) {
+        objectManager.delete(account);
```

Comment:
In account *purge* you need a process handler (extend AbstractDefaultProcessHandler and call it AccountPurge) that does:

Deactivate then remove all hosts.

Create this as a generic "AccountCleanup" handler.  Which will continue to add more account cleanup operations to.